src/main/scala/com/gd/twitteranalytics/reports/ActiveUserDatasetReport.scala

```diff
@@ -0,0 +1,33 @@
+package com.gd.twitteranalytics.reports
+
+import ActiveUserReportProcessor._
+import TwitterReportGenerator._
+import TwitterReportWriter._
+import org.apache.hadoop.fs.{FileSystem, Path}
+
+object ActiveUserDatasetReport extends TwitterReport{
+  val log = getLogger("ActiveUserDataSetReport")
+
+  def main(args: Array[String]): Unit = {
+    val spark = setSparkSession("TwitterActiveUserDataSetReport")
+    val fs = FileSystem.get(spark.sparkContext.hadoopConfiguration)
+
+    if (fs.exists(new Path(dataPath))) {
+      log.debug("=======> Building Twitter ActiveUser Report for date " + today)
```

Comment:
for backfilling or re-processing purposes, it should be possible to run a report for any particular date in the past.

Reply:
Okay.. so current logic is written on the requirement that report will be run on a daily basis on a daily data... To implement reporting for re-processing purpose, how we are supposed to get past date on which report is to be run ? Through Config ?

Reply:
> report will be run on a daily basis on a daily data...

it implies a scheduling requirement.
In general, backfilling and re-processing activities are parts of data engineering process, that I assumes you was familiar with.

Reply:
> how we are supposed to get past date on which report is to be run ? Through Config ?

usually it's done via arguments or config, yes.